src/filter.ts

```diff
@@ -1,4 +1,4 @@
-import { FilterFuncion } from './functions';
+import type { FilterFuncion } from './functions';
```

Comment:
Wow, I didn't know about this feature.

By the way, why do we need to use type imports? We compile the module to JS, so there're no types in dist files, even if we use "normal" import.

Reply:
It aids the typescript compiler in knowing when imports can be fully elided. The compiler can't accurately know that there are no side-effects in `functions.ts`, so this:

```
import { FilterFuncion } from './functions';
```

Gets transpiled to this:

```
import './functions';
```

Even though there's no reason to be importing that module. By using `import type`, the compiler knows for certain it can fully elide that import statement.

Reply:
It's probably worth mentioning that I didn't do a comprehensive review of the code to check for other possible places where `import type` can be used, although I'm happy to do that in this PR or another PR if you'd like.

Reply:
Interesting. Thanks for explaining this.